nautobot/extras/health_checks.py

```diff
@@ -26,3 +28,34 @@ def check_status(self):
             raise ServiceReturnedUnexpectedResult("Integrity Error")
         except DatabaseError:
             raise ServiceUnavailable("Database error")
+
+
+class RedisBackend(BaseHealthCheckBackend):
+    """Health check for Redis."""
+
+    redis_url = getattr(settings, "CACHEOPS_REDIS", "redis://localhost:6379/1")
+    sentinel_url = getattr(settings, "CACHEOPS_SENTINEL", None)
+
+    def check_status(self):
+        """Check Redis service by pinging the redis instance with a redis connection."""
+        # if CACHEOPS_REDIS is set to False or Sentinel is enabled
+        # We need to check Redis using Sentinel
+        if not self.redis_url or self.sentinel_url is not None:
+            try:
+                sentinel = Sentinel(self.sentinel_url["locations"], socket_timeout=self.sentinel_url["socket_timeout"])
+                with sentinel.master_for(self.sentinel_url["service_name"], db=self.sentinel_url["db"]) as master:
+                    master.ping()
+            except (ConnectionRefusedError, exceptions.ConnectionError, exceptions.TimeoutError) as e:
+                self.add_error(ServiceUnavailable("Unable to connect to Redis Sentinel: %s" % type(e).__name__), e)
+            except BaseException as e:
```

Comment:
`BaseException` includes things like `SystemExit` and `KeyboardInterrupt` that we probably don't need or want to catch. I think just catching `Exception` here and below would be preferable.

Reply:
@glennmatthews I've replaced `BaseException` by `Exception` but it will not change anything because the parent class is using `BaseException` : https://github.com/KristianOellegaard/django-health-check/blob/1807ffe7d166acdc11262adc36eff6efe5e2836a/health_check/backends.py#L33